src/configuration/remapper.ts

```diff
@@ -76,29 +88,194 @@ export class Remapper implements IRemapper {
     vimState: VimState
   ): Promise<boolean> {
     this._isPotentialRemap = false;
+    this._allowPotentialRemapOnFirstKey = vimState.recordedState.allowPotentialRemapOnFirstKey;
+    let allowBufferingKeys = true;
+    let remainingKeys: string[] = [];
 
     if (!this._remappedModes.includes(vimState.currentMode)) {
       return false;
     }
 
     const userDefinedRemappings = configuration[this._configKey] as Map<string, IKeyRemapping>;
 
+    if (keys[keys.length - 1] === '<BufferedKeys>') {
+      // Timeout finished. Don't let an ambiguous or potential remap start another timeout again
+      keys = keys.slice(0, keys.length - 1);
```

Comment:
I think `pop()` is clearer here

Reply:
This needs to be a copy because the other remapper will still be called with the same list and we can't change it.

Reply:
Ahh, I see, thanks